File: String/59palindromicSubstring/palindromicSubstring.java

public class Solution {
    public static int countPalindrome(int l, int r, String s) {
        int res = 0;
        while (l >= 0 && r < s.length() && s.charAt(l) == s.charAt(r)) {
            res += 1;
            l -= 1;
            r += 1;
        }
        return res;
    }

    public static int isPalindromeSubSt(String s) {
        int res = 0;
        for (int idx = 0; idx < s.length(); idx++) {
            res += countPalindrome(idx, idx, s);
            res += countPalindrome(idx, idx + 1, s);
        }
        return res;
    }

    public static void main(String[] args) {
        System.out.println(isPalindromeSubSt("aaab"));
        System.out.println(isPalindromeSubSt("aaa"));
        System.out.println(isPalindromeSubSt("abc"));
    }
}

File: String/59palindromicSubstring/palindromicSubstring.py
# leetcode 647 palindromic substring
class Solution:
    def isPalindromicSubSt(self, s: str) -> int:
        # O(n**2) time complexity
        res = 0
        for i in range(len(s)):
            res += self.countPalindrome(i, i , s)
            res += self.countPalindrome(i, i + 1, s)
        return res
    def countPalindrome(self, l: int, r: int, s: str) -> int:
        res = 0
        while l >= 0 and r < len(s) and s[l] == s[r]:
            res += 1
            l -= 1
            r += 1
        return res
print(Solution().isPalindromicSubSt("aaab"))
print(Solution().isPalindromicSubSt("aaa"))
print(Solution().isPalindromicSubSt("abc"))